lib/repeatable.js

```diff
@@ -34,33 +39,44 @@ module.exports = function(Queue) {
 
       nextMillis = nextMillis.getTime();
 
+      if (skipCheckExists) return createNextJob();
+
       return client
-        .zadd(_this.keys.repeat, nextMillis, repeatJobKey)
-        .then(function() {
-          //
-          // Generate unique job id for this iteration.
-          //
-          var customId = getRepeatJobId(
-            name,
-            jobId,
-            nextMillis,
-            md5(repeatJobKey)
-          );
-          now = Date.now();
-          var delay = nextMillis - now;
-
-          return Job.create(
-            _this,
-            name,
-            data,
-            _.extend(_.clone(opts), {
-              jobId: customId,
-              delay: delay < 0 ? 0 : delay,
-              timestamp: now,
-              prevMillis: nextMillis
-            })
-          );
+        .zscore(_this.keys.repeat, repeatJobKey)
+        .then(function handleZScore(repeatableExists) {
```

Comment:
although better than before, there is still a hazard here, since the job could have been deleted after the call to zscore and before the call to create a new job. The only way to fix this correctly is using a LUA script... thats much more job, I think we should at least comment this fact so that it can be fixed in the future.